src/compiler/tools/LanguageService.ts

```diff
@@ -337,6 +340,41 @@ export class LanguageService {
             .map(fileTextChanges => new FileTextChanges(fileTextChanges));
     }
 
+    /**
+     * Gets the edits information to modify the file with given refactor action at given position.
+     * @param fileName - File path of the source file.
+     * @param formatOptions - Fomat code settings.
+     * @param positionOrRange - The position in the source file where to apply given refactor.
+     * @param refactorName  - The refactor name.
+     * @param actionName  - The refactor action name.
+     * @param preferences - User preferences for refactoring.
+     */
+    getEditsForRefactor(fileName: string, formatOptions: FormatCodeSettings, positionOrRange: number | TextRange,
```

Comment:
What do you think about making this a single object parameter with all of these parameters as properties on the object?

Reply:
(and same for `getCodeFixesAtPosition`)

Reply:
> What do you think about making this a single object parameter with all of these parameters as properties on the object?

I think having a similar API to TypeScript will facilitate things for users, so my opinion is no :)